code/src/HC_SR04_c.cpp

```diff
@@ -0,0 +1,24 @@
+#include "HC_SR04_c.hpp"
+
+int16_t R2D2::Distance::HC_SR04_c::get_distance(){
+    trigger_pin.write(0);
+    hwlib::wait_us(2);
+    trigger_pin.write(1);
+    hwlib::wait_us(10);
+    trigger_pin.write(0);
+
+    while(!echo_pin.read()){
+        hwlib::wait_us(10);
+    }
+
+    uint32_t ticks_start = hwlib::now_ticks();
+
+    while(echo_pin.read()){
+        hwlib::wait_us(10);
```

Comment:
Same here, I think the measurements would be more more accurate without the hwlib::wait.

Reply:
I have tested the accuracy it works well and the wait is for a future RTOS.